src/components/TextInput.js

```diff
@@ -9,13 +9,12 @@ import classnames from 'classnames';
  *   the rendered `input` element.
  * @prop {boolean} [hasError] - There is an error associated with this input. Will
  *   set some error styling.
+ * @prop {'email'|'search'|'text'|'url'} [type="text"] - Set the <input> type:
```

Comment:
We could add 'password' too (while not having a specific component for that).

Something to notice about the 'search' is that in Chrome it adds a 'X' cancel symbol. I am not sure if we mind that:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search#differences_between_search_and_text_types